README.md

```diff
@@ -1,2 +1,76 @@
-# cmo-metadb-label-generator
-CMO MetaDB messaging service for handling CMO label generation
+# CMO Label Generator 🔍
+
+CMO Label Generator is a distributed microservices system. It receives messages from LIMS when a request is marked delivered. This message is processed and a unique CMO label Id is generated based on specifications that can be found in CMOPatientIDandSampleIDgeneration.pdf. Any new message is then published to downstream subscribers. The message published is a JSON reqpresentation of MetaDbRequest (more details can be found in cmo-metadb repository). 
+
+Label Generator is dependent on MetaDb, Nats Jetstream and Nats RequestReply.
+
+## Run
+
+### Custom properties
+
+Make an `application.properties` based on [application.properties.EXAMPLE](src/main/resources/application.properties.EXAMPLE).
+
+- `igo.cmo_label_generator_topic` : receives requests to process
+- `request_reply.patient_samples_topic` : used to patient samples set from MetaDB
+- `igo.new_request_topic` : publishes the processed request to this topic
+
+All properties are required with the exception of some NATS connection-specific properties. The following are only required if `nats.tls_channel` is set to `true`:
+
+- `nats.keystore_path` : path to client keystore
```

Comment:
Add a sub header above these props to indicate that they are related to NATS connection props